experimental/SetPartitions/src/LinearSetPartition.jl

```diff
@@ -0,0 +1,235 @@
+```
+"""
+function compose(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}, d::T) where 
+        { S <: AbstractPartition, T <: RingElement }
+    result = Dict{S, T}()
+    
+    for i in pairs(coefficients(p))
+        for ii in pairs(coefficients(q))
+            (composition, loop) = compose_count_loops(i[1], ii[1])
+            new_coefficient = i[2] * ii[2] * (d^loop)
+            result[composition] = get(result, composition, 0) + new_coefficient
+        end
+    end
+    return linear_partition(result)
+end 
+
+"""
+    tensor_product(p::LinearSetPartition{P, R}, q::LinearSetPartition{P, R}) where 
+        { P <: AbstractPartition, R <: RingElement }
+
+Return the tensor product of `p` and `q`.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in d over QQ, d)
+
+julia> a = linear_partition([(set_partition([1, 2], [1, 1]), S(4)), (set_partition([1, 1], [1, 1]), 4*d)])
+LinearSetPartition{SetPartition, QQPolyRingElem}(Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 2], [1, 1]) => 4, SetPartition([1, 1], [1, 1]) => 4*d))
+
+julia> tensor_product(a, a)
+LinearSetPartition{SetPartition, QQPolyRingElem}(Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 1, 2, 2], [1, 1, 2, 2]) => 16*d^2, SetPartition([1, 2, 3, 3], [1, 1, 3, 3]) => 16*d, SetPartition([1, 2, 3, 4], [1, 1, 3, 3]) => 16, SetPartition([1, 1, 2, 3], [1, 1, 2, 2]) => 16*d))
+```
+"""
+function tensor_product(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+
+    result = Dict{S, T}()
+    
+    for i in pairs(coefficients(p))
+        for ii in pairs(coefficients(q))
+            composition = tensor_product(i[1], ii[1])
+            new_coefficient = i[2] * ii[2]
+            result[composition] = get(result, composition, 0) + new_coefficient
+        end
+    end
+
+    return linear_partition(result)
+end 
+
+function ⊗(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+    return tensor_product(p, q)
+end
+
+"""
+    -(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+
+Perform a subtraction between `p` and `q` and return the result.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in d over QQ, d)
+
+julia> a = linear_partition([(set_partition([1, 2], [1, 1]), S(4)), (set_partition([1, 1], [1, 1]), 4*d)])
+LinearSetPartition{SetPartition, QQPolyRingElem}(Dict{SetPartition, QQPolyRingElem}(SetPartition([1, 2], [1, 1]) => 4, SetPartition([1, 1], [1, 1]) => 4*d))
+
+julia> a - a
+LinearSetPartition{SetPartition, QQPolyRingElem}(Dict{SetPartition, QQPolyRingElem}())
+```
+"""
+function -(p::LinearSetPartition{S, T}, q::LinearSetPartition{S, T}) where 
+        { S <: AbstractPartition, T <: RingElement }
+    return p + scale(-1, q)
+end
```

Comment:
What about adding a unary - method? This could then of course be used here as `p + (-q)`